### PR TITLE
Xubz

### PR DESCRIPTION
**Pull request by Zubayr-Imkaraaz**
publication controller updated

### CODE REVIEW

File: App/models/publication.py [hunk 1]
```diff
@@ -8,6 +8,8 @@ class Publication(db.Model):
     title = db.Column(db.String(120), nullable=False, unique=True)
     authorId = db.Column(db.Integer, db.ForeignKey("author.id"), nullable=False)
     coauthors = db.relationship("Author", secondary=CoAuthorPublication)
+    abstract = db.Column(db.String(200), nullable=False, unique=True)
+    dateOfPublication = db.Column(db.Date, nullable=False, unique=True)
 
     def __init__(self, title, authorId, coauthors):
```
Comment (crowbar72) [inline App/models/publication.py]: This is the function that is called when a new Publication is created. This needs to be changed to accept an abstract and dateOfPublication and set each of them from the data it takes in.

File: App/controllers/user.py [hunk 2]
```diff
@@ -3,8 +3,14 @@
 
 def create_user(username, password, authorId):
     newuser = User(username=username, password=password, authorId=authorId)
+    author = Author.query.filter_by(name=name).first()
+    author.id = authorId
```
Comment (crowbar72) [inline App/controllers/user.py:8]: This never commits, but what is it supposed to do?

**Review by crowbar72 — CHANGES_REQUESTED** [review]

File: App/models/publication.py (hunk 1 above)
Comment (Zubayr-Imkaraaz) [reply App/models/publication.py]: the controller was updated as well to accept the 2 new additions

**Comment by Zubayr-Imkaraaz** [comment]
added name to user table to compare with name in auth table. create user now inputs a name as well to query with and set author id to new author id when the user is created

**Review by crowbar72 — APPROVED** [review]

**Pull request merged**